src/main/java/org/amahi/anywhere/util/Downloader.java

```diff
@@ -115,12 +117,15 @@ private void finishDownloading() {
             String downloadUri = downloadInformation.getString(
                 downloadInformation.getColumnIndex(DownloadManager.COLUMN_LOCAL_URI));
 
-            if (downloadUri.substring(0, 7).matches("file://")) {
-                downloadUri = downloadUri.substring(7);
+            try {
+                URI uri = new URI(downloadUri);
+                downloadUri = uri.getPath();
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
```

Comment:
minor niggle. wouldn't we want to print the URI, tho, not the stack trace?

Reply:
Stack trace contains the URI, but I too feel that printing URI only will work fine plus there is one more change that I think is necessary (`/file:` need to be removed from the Uri), so I will update the PR.